contrib/platform/src/com/sun/jna/platform/win32/COM/COMException.java

```diff
@@ -35,93 +33,74 @@
-
-    /** The pu arg err. */
-    private IntByReference puArgErr;
-
-    private int uArgErr;
+    private final HRESULT hresult;
 
     /**
      * Instantiates a new automation exception.
      */
     public COMException() {
-        super();
+        this("", (Throwable) null);
     }
 
     /**
      * Instantiates a new automation exception.
      *
      * @param message
      *            the message
-     * @param cause
-     *            the cause
      */
-    public COMException(String message, Throwable cause) {
-        super(message, cause);
+    public COMException(String message) {
+        this(message, (Throwable) null);
     }
 
+
     /**
      * Instantiates a new automation exception.
      *
-     * @param message
-     *            the message
+     * @param cause
+     *            the cause
      */
-    public COMException(String message) {
-        super(message);
+    public COMException(Throwable cause) {
+        this(null, cause);
     }
 
     /**
      * Instantiates a new automation exception.
      *
      * @param message
      *            the message
-     * @param pExcepInfo
-     *            the excep info
-     * @param puArgErr
-     *            the pu arg err
+     * @param cause
+     *            the cause
      */
-    public COMException(String message, EXCEPINFO pExcepInfo,
-            IntByReference puArgErr) {
-        super(message + " (puArgErr=" + (null==puArgErr?"":puArgErr.getValue()) + ")");
-        this.pExcepInfo = pExcepInfo;
-        this.puArgErr = puArgErr;
+    public COMException(String message, Throwable cause) {
+        super(message, cause);
+        this.hresult = null;
     }
 
     /**
      * Instantiates a new automation exception.
      *
-     * @param cause
-     *            the cause
+     * @param message
+     *            the message
+     * @param hresult
+     *            HRESULT that lead to the creation of the COMException
      */
-    public COMException(Throwable cause) {
-        super(cause);
+    public COMException(String message, HRESULT hresult) {
+        super(message);
+        this.hresult = hresult;
     }
 
     /**
-     * Gets the excep info.
-     *
-     * @return the excep info
+     * @return the HRESULT that lead to thie COMException or NULL if the COMException as not directly caused by a native call
      */
-    public EXCEPINFO getExcepInfo() {
-        return pExcepInfo;
+    public HRESULT getHresult() {
+        return hresult;
     }
 
     /**
-     * Gets the arg err.
-     *
-     * @return the arg err
+     * @param errorToCheck
+     * @return true if the exception has an associated HRESULT and that HRESULT matches the supplied value
      */
-    public IntByReference getArgErr() {
-        return puArgErr;
-    }
-
-    public int getuArgErr() {
-        return uArgErr;
-    }
-
-    public void setuArgErr(int uArgErr) {
-        this.uArgErr = uArgErr;
+    public boolean matchesHresult(int errorToCheck) {
```

Comment:
This seems to imply that the argument is the Hresult. Should be `matches` or `matchesErrorCode`.